core/src/main/java/org/springframework/security/oauth2/server/authorization/OAuth2Authorization.java

```diff
@@ -15,17 +15,108 @@
  */
 package org.springframework.security.oauth2.server.authorization;
 
+import org.springframework.lang.Nullable;
 import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.util.Assert;
 
+import java.util.Collections;
 import java.util.Map;
+import java.util.Objects;
 
 /**
+ * Represents a collection of attributes which describe an OAuth 2.0 authorization context.
+ *
  * @author Joe Grandja
+ * @author Krisztian Toth
  */
 public class OAuth2Authorization {
-	private String registeredClientId;
-	private String principalName;
-	private OAuth2AccessToken accessToken;
-	private Map<String, Object> attributes;
+	private final String registeredClientId;
+	private final String principalName;
+	private final OAuth2AccessToken accessToken;
+	private final Map<String, Object> attributes;
+
+	/**
+	 * Creates a {@link OAuth2Authorization} object with the provided params and an empty, unmodifiable
+	 * {@code attributes} map.
+	 *
+	 * @see #OAuth2Authorization(String, String, OAuth2AccessToken, Map)
+	 */
+	public OAuth2Authorization(String registeredClientId, String principalName, OAuth2AccessToken accessToken) {
+		this(registeredClientId, principalName, accessToken, Collections.emptyMap());
+	}
+
+	/**
+	 * Creates a {@link OAuth2Authorization} object without an {@link OAuth2AccessToken}.
+	 *
+	 * @see #OAuth2Authorization(String, String, OAuth2AccessToken, Map)
+	 */
+	public OAuth2Authorization(String registeredClientId, String principalName, Map<String, Object> attributes) {
```

Comment:
Instead of overloaded constructors, lets a `Builder`. Take a look at the recent merge of `RegisteredClient` and apply the same `Builder` pattern here.

Reply:
Added Builder in `9b748c4`